pkg/issuer/acme/dns/dns.go

```diff
@@ -52,6 +54,7 @@ type Solver struct {
 	resourceNamespace       string
 	dnsProviderConstructors dnsProviderConstructors
 	ambientCredentials      bool
+	dnsCheckMethod          DNSCheckMethod
```

Comment:
I think it feels a bit more natural in go to invert this and instead of having a check method type here that gets plumbed into the `checkDNSPropagation` method, instead to have a "DNSChecker" interface that can "LookupDNS" or such, and then have two implementations of that interface.

I think that's a bit awkward with the code as it is because of how many package-level functions are called for dns lookups, but they probably shouldn't be package level

Reply:
DNS over https supports DNS binary packets (https://developers.cloudflare.com/1.1.1.1/dns-over-https/wireformat/) , so probably it is best to wait for  https://github.com/jetstack/cert-manager/pull/710 to be merged and then extend accepted values to accept URLs, then if URL is passed just change the way UDP packets are sent